src/Microsoft.AspNetCore.Http.Abstractions/HttpRequest.cs

```diff
@@ -108,6 +109,7 @@ public abstract class HttpRequest
         /// <summary>
         /// Gets or sets the request body as a form.
         /// </summary>
+        [Obsolete("This is obsolete and will be removed in a future version. The recommended alternative is to use Microsoft.AspNetCore.Http.HttpRequest.ReadFormAsync(). See TODO")]
         public abstract IFormCollection Form { get; set; }
```

Comment:
Make sure you address consumers of this in other repos:

* https://github.com/aspnet/Antiforgery/blob/dev/test/Microsoft.AspNetCore.Antiforgery.Test/Internal/DefaultAntiforgeryTokenStoreTest.cs#L83
* https://github.com/aspnet/Identity/blob/dev/test/Microsoft.AspNetCore.Identity.Service.Core.Test/AuthorizationCodeExchangeIntegrationTest.cs#L35
* https://github.com/aspnet/Entropy/blob/dev/samples/Content.Upload.Form/Startup.cs#L23 (this one's a WTF)
* https://github.com/aspnet/MusicStore/blob/dev/test/MusicStore.Test/CheckoutControllerTest.cs#L65
* https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.IntegrationTests/CollectionModelBinderIntegrationTest.cs#L602
etc